tests/behaviour/cli.feature

```diff
@@ -7,4 +7,64 @@ Feature: Test the csvcubed Command Line Interface.
       """
       Publisher 'https://im-not-a-real-publisher.com' is not recognised by csvcubed.
       """
-    Then remove test log files
\ No newline at end of file
+    Then remove test log files
+
+  Scenario: The csvcubed code-list build command should output a code-list CSVW without the need to provide a CSV
+    Given the existing test-case file "readers/code-list-config/v1.0/code_list_config_hierarchical.json"
+    When the csvcubed CLI is run with "code-list build readers/code-list-config/v1.0/code_list_config_hierarchical.json --validation-errors-to-file"
+    Then the file at "out/title-of-the-code-list.csv" should exist
+    And the file at "out/title-of-the-code-list.csv-metadata.json" should exist
+    And the file at "out/title-of-the-code-list.table.json" should exist
+    And the file at "out/validation-errors.json" should exist
+    And the validation-errors.json file should contain
+    """
+    "offending_value": "http://purl.org/dc/aboutdcmi#DCMI"
+    """
+    Then remove test log files
+
+  Scenario: The csvcubed code-list build command should display the logging in accordance with the log level set at debug.
+    Given the existing test-case file "readers/code-list-config/v1.0/code_list_config_produces_error.json"
+    When the csvcubed CLI is run with "code-list build readers/code-list-config/v1.0/code_list_config_produces_error.json --log-level debug"
+    Then the command line output should display the log message
+    """
+    csvcubed.utils.cli - WARNING - Schema Validation Error: $.creator - 'http://purl.org/dc/aboutdcmi#DCMI' is not one of ['http://dbpedia.org/resource/Open_Knowledge_Foundation', 'http://statistics.data.gov.uk', 'https:/
+    """
+    And the command line output should display the log message
+    """
+    csvcubed.utils.cli - ERROR - Validation Error: An error was encountered when validating the cube. The error occurred in '["('concepts', 3)", 'existing_concept_uri']' and was reported as ''This will produce an error' does not look like a URI.'
```

Comment:
Can you match on something shorter than this? Make sure not to include the path/namespace of the python module as this might change if we refactor code.

Reply:
Also, I don't think we need to test that every level of logging above DEBUG is printed. I'm sure the logging library has tests for this.

Reply:
done

Reply:
I am not sure about the logging library. Might need some more details on it please